File: replace_images.py
import os
import glob
import re

# Get all tsx files in src/app/(site)/
files = glob.glob(r'c:\HandsOn\handson-app\src\app\(site)\**\page.tsx', recursive=True)

for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original_content = content

    # Add import Image if not present and if there's an img tag
    if '<img' in content and 'import Image' not in content:
        # Find the first import and add it before
        if 'import' in content:
            content = re.sub(r'^(import.*?)$', r"import Image from 'next/image'\n\1", content, count=1, flags=re.MULTILINE)
        else:
            content = "import Image from 'next/image'\n\n" + content
    
    # Replace <img src="..." alt="..." className="..."> with <Image src="..." alt="..." className="..." width={1200} height={800} />
    def replace_img(match):
        attrs = match.group(1)
        # Check if width/height are already present
        if 'width=' not in attrs:
            attrs += ' width={1200} height={800}'
        return f"<Image{attrs} />"

    content = re.sub(r'<img\s+([^>]+?)\s*/?>', replace_img, content)

    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

print("Done")

File: replace_images.mjs
import fs from 'fs'
import path from 'path'
import { globSync } from 'glob' // Next.js projects usually have glob or I can just write a recursive readdir

const dir = './src/app/(site)'

function getAllFiles(dirPath, arrayOfFiles) {
  const files = fs.readdirSync(dirPath)
  arrayOfFiles = arrayOfFiles || []

  files.forEach(function(file) {
    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
      arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles)
    } else {
      if (file.endsWith('.tsx')) {
        arrayOfFiles.push(path.join(__dirname, dirPath, "/", file))
      }
    }
  })

  return arrayOfFiles
}

const files = getAllFiles(dir, [])

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8')
  const original = content

  if (content.includes('<img') && !content.includes('import Image')) {
    if (content.includes('import ')) {
      content = content.replace(/^(import.*)$/m, "import Image from 'next/image'\n$1")
    } else {
      content = "import Image from 'next/image'\n\n" + content
    }
  }

  content = content.replace(/<img\s+([^>]+?)\s*\/?>/g, (match, attrs) => {
    if (!attrs.includes('width=')) {
      attrs += ' width={1200} height={800}'
    }
    return `<Image ${attrs.trim()} />`
  })

  if (content !== original) {
    fs.writeFileSync(file, content, 'utf8')
    console.log('Updated ' + file)
  }
})
